animata/card/survey-card.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useRef, useState } from "react";
+
+import { cn } from "@/lib/utils";
+
+interface SurveyCardProps {
+  /**
+   * The items to display in the Survey.
+   * Each item should have a vote and itemName.
+   */
+  items: {
+    vote: number;
+    itemName?: string;
+  }[];
+  /**
+   * The width of the Survey. recommended to use with more than 250px
+   */
+  width?: number;
+  /**
+   * The title of the Survey.
+   */
+  surveyTitle?: string;
+}
+
+export default function SurveyCard({ items, width: providedWidth, surveyTitle }: SurveyCardProps) {
+  const [{ width }, setSize] = useState({
+    width: providedWidth ?? 250,
+  });
+  const [totalVotes, setTotalVotes] = useState(0);
+  const [maxVote, setMaxVote] = useState(0);
```

Comment:
You don't need state for these, you can use `useMemo` or simply calculate it inline.